Assignment2Q2.java

```diff
@@ -0,0 +1,129 @@
+
+public class Assignment2Q2 {
+	public static void main(String[] args) {
+		// construct the tree
+		BinaryTreeNode<Integer> n1 = new BinaryTreeNode<>(1, null, null);
+		BinaryTreeNode<Integer> n2 = new BinaryTreeNode<>(5, null, null);
+		BinaryTreeNode<Integer> n3 = new BinaryTreeNode<>(3, n1, n2);
+		BinaryTreeNode<Integer> n4 = new BinaryTreeNode<>(14, null, null);
+		BinaryTreeNode<Integer> n5 = new BinaryTreeNode<>(9, n3, n4);
+		BinaryTreeNode<Integer> n6 = new BinaryTreeNode<>(19, null, null);
+		BinaryTreeNode<Integer> n7 = new BinaryTreeNode<>(18, null, n6);
+		BinaryTreeNode<Integer> n8 = new BinaryTreeNode<>(16, n5, n7);
+		BinaryTreeNode<Integer> n9 = new BinaryTreeNode<>(186, null, null);
+
+		System.out.println(findCommonAncestors(n9, n1, n8).getData());
+	}
+
+	/**
+	 * Finds lowest common ancestor of node1 and node2
+	 * 
+	 * @param node1
+	 *            first node
+	 * @param node2
+	 *            second node
+	 * @param root
+	 *            root of the tree
+	 * @return lowest common ancestor of node1 and node2
+	 * @throws IllegalArgumentException
+	 *             if one or both of the nodes are not present in the tree
+	 */
+	public static <T> BinaryTreeNode<T> findCommonAncestors(BinaryTreeNode<T> node1, BinaryTreeNode<T> node2,

```

Comment:
I first thought of that, but to use the code for findAncestors, I will have to store them in a list, which we are not allowed to do. That's why I did it like this.